lib/tasks/mastodon.rake

```diff
@@ -395,7 +395,7 @@ namespace :mastodon do
         incompatible_syntax = false
 
         env_contents = env.each_pair.map do |key, value|
-          if value.is_a?(String) && value =~ /[\s\#\\"]/
+          if value.is_a?(String) && value =~ /[\s\#\\"\$]/
```

Comment:
Instead of trying to fix this manually, should we use `Shellwords.escape`

Reply:
Possibly, but IIRC `dotenv` doesn't *exactly* implement shell rules.

Reply:
Yeah it still behaves the same. `Shellwords.escape('foo!#test')` would return `'foo\!\#test'`, which would be parsed by dotenv as `'foo!\'`.
That being said we probably can do something as simple as `"\"#{Shellwords.escape(value)}\""`.

Reply:
From `Shellwords::shellescape`'s documentation:
> Note that a resulted string should be used unquoted and is not intended for use in double quotes nor in single quotes.

And indeed, this would result in double-escaping some characters…

I'm going to more thoroughly check Dotenv's behavior and better document that code.

Reply:
Wrote a documented escape function that I tested to be fairly robust. I don't think *everything* can be escaped, though.